### PR TITLE
only set upstreamTsContext.AlpnProtocols to 'istio' in ISTIO_MUTUAL mode

### PR DESCRIPTION
**Pull request by quanjielin**
context: https://github.com/istio/istio/issues/6928

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -477,8 +477,12 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 		}
 		if cluster.Http2ProtocolOptions != nil {
 			// This is HTTP/2 in-mesh cluster, advertise it with ALPN.
-			cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNInMeshH2
-		} else {
+			if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL {
+				cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNInMeshH2
+			} else {
+				cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNH2Only
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: I don't think we can say ALPNH2only when non istio mutual TLS is used. We need to allow h2 and http/1.1. i.e. the same stuff as the ListenerALPNProtocols

Comment (rshriram) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: Oh ignore.. I see the cluster.Http2ProtocolOptions if check above..

**Comment by rshriram** [comment]
Rebase to release-1.0 branch

**Comment by hklai** [comment]
@quanjielin looks like this is needed in 1.0. The easiest way is to click on the edit button next to the PR title, and then change the base from master to release-1.0.

**Comment by quanjielin** [comment]
rebase got messed up(took master changes yesterday), will close this one and open another one.

**Pull request closed without merging**